mindsdb/integrations/handlers/stabilityai_handler/stabilityai.py

```diff
@@ -0,0 +1,79 @@
+from stability_sdk import client
+from stability_sdk.client import generation
+from PIL import Image
+import io, requests
+
+class StabilityAPIClient:
+    
+    def __init__(self, api_key, dir_to_save, 
+                 engine="stable-diffusion-xl-1024-v1-0", upscale_engine="esrgan-v1-x2plus"):
+        self.api_key = api_key
+        self.STABILITY_HOST = 'grpc.stability.ai:443'
+        self.save_dir = dir_to_save + "/" if not dir_to_save.endswith("/") else dir_to_save
+        self.available_engines = self.get_existing_engines()
+        if not self._is_valid_engine(engine):
+            raise ValueError("Unknown engine. The available engines are - " + self.available_engines)
+        if not self._is_valid_engine(upscale_engine):
+            raise ValueError("Unknown upscale engine. The available engines are - " + self.available_engines)
+        
+        self.stability_api = client.StabilityInference(host=self.STABILITY_HOST, 
+                                                       key=self.api_key,
+                                                       engine=engine,
+                                                       upscale_engine=upscale_engine,
+                                                       verbose=True)
+        
+    def save_image(self, artifact):
+        img = Image.open(io.BytesIO(artifact.binary))
```

Comment:
Please mention in the comments about what this function does.

Reply:
Done